janusgraph-hadoop-parent/janusgraph-hadoop-core/src/main/java/org/janusgraph/hadoop/formats/cql/CqlBinaryInputFormat.java

```diff
@@ -0,0 +1,97 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.janusgraph.hadoop.formats.cql;
+
+import org.apache.cassandra.hadoop.ConfigHelper;
+import org.apache.cassandra.hadoop.cql3.CqlInputFormat;
+import org.apache.cassandra.hadoop.cql3.CqlRecordReader;
+import org.apache.cassandra.thrift.SlicePredicate;
+import org.apache.cassandra.thrift.SliceRange;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.RecordReader;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.janusgraph.diskstorage.Entry;
+import org.janusgraph.diskstorage.StaticBuffer;
+import org.janusgraph.diskstorage.cql.CQLConfigOptions;
+import org.janusgraph.graphdb.configuration.GraphDatabaseConfiguration;
+import org.janusgraph.hadoop.config.JanusGraphHadoopConfiguration;
+import org.janusgraph.hadoop.formats.util.AbstractBinaryInputFormat;
+import org.janusgraph.hadoop.formats.util.input.JanusGraphHadoopSetupCommon;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.List;
+
+public class CqlBinaryInputFormat extends AbstractBinaryInputFormat {
+
+    private static final Logger log = LoggerFactory.getLogger(CqlBinaryInputFormat.class);
+
+    // Copied these private constants from Cassandra's ConfigHelper circa 2.0.9
+    private static final String INPUT_WIDEROWS_CONFIG = "cassandra.input.widerows";
+    private static final String RANGE_BATCH_SIZE_CONFIG = "cassandra.range.batch.size";
+
+    private final CqlInputFormat  cqlInputFormat = new CqlInputFormat();
+
+    @Override
+    public List<InputSplit> getSplits(JobContext jobContext) throws IOException {
+        return cqlInputFormat.getSplits(jobContext);
+    }
+
+    @Override
+    public RecordReader<StaticBuffer, Iterable<Entry>> createRecordReader(InputSplit inputSplit, TaskAttemptContext taskAttemptContext) throws IOException, InterruptedException {
+        CqlRecordReader recordReader = (CqlRecordReader) cqlInputFormat.createRecordReader(inputSplit, taskAttemptContext);
+        CqlBinaryRecordReader reader = new CqlBinaryRecordReader(recordReader);
+
+        return reader;
+    }
+
+    @Override
+    public void setConf(final Configuration config) {
+        super.setConf(config);
+
+        // Copy some JanusGraph configuration keys to the Hadoop Configuration keys used by Cassandra's ColumnFamilyInputFormat
+        ConfigHelper.setInputInitialAddress(config, janusgraphConf.get(GraphDatabaseConfiguration.STORAGE_HOSTS)[0]);
```

Comment:
@farodin91 @FlorianHockmann Why only the first host? If the first host is down or unresponsive for some time, the whole MapReduce or spark job would fail.

Reply:
We used the Thrift implementation as a basis and just changed the parts that needed to be changed for CQL. Here's the same line in the `CassandraBinaryInputFormat`:
https://github.com/JanusGraph/janusgraph/blob/456eff043e1bd14bb6713d1025dcb006aca60be0/janusgraph-hadoop-parent/janusgraph-hadoop-core/src/main/java/org/janusgraph/hadoop/formats/cassandra/CassandraBinaryInputFormat.java#L76

and that line was already there in the old Titan implementation. 

But yes, it probably makes sense to make this more robust.

Reply:
Thanks for the quick heads-up! We just made the change locally and will test it for some time. If it works fine I'll submit a PR to make it more robust.